eclipse/build.gradle.kts

```diff
@@ -58,4 +63,131 @@ tasks {
+        }
+        from("update/site.xml")
+        into(updateSiteDirPath)
+    }
+
+    /* Step 1 of update-site creation
+     *
+     * Creates the basic p2-Repository, but it is
+     * not usable as update-site, because the plugins
+     * are not visible to users.
+     *
+     * equivalent to P2Director call with:
+     * <code>
+     * org.eclipse.equinox.p2.publisher.FeaturesAndBundlesPublisher \
+     *   -source <project dir>/build/update-site \
+     *   -compress \
+     *   -inplace \
+     *   -append \
+     *   -publishArtifacts
+     * </code>
+     */
+    val updateSiteFeaturesAndBundlesPublishing by registering {
+        dependsOn(updateSitePreparation)
+        doLast {
+            with(com.diffplug.gradle.p2.FeaturesAndBundlesPublisher()) {
+                source(project.file(updateSiteDirPath))
+                compress()
+                inplace()
+                append()
+                publishArtifacts()
+                runUsingBootstrapper()
+            }
+        }
+    }
+
+    /* Step 2 of update-site creation
+     *
+     * Adds the meta-data to the repository to make
+     * the plugin accessible for users.
+     *
+     * equivalent to P2Director call with:
+     * <code>
+     * org.eclipse.equinox.p2.publisher.CategoryPublisher \
+     *   -metadataRepository file:<project dir>/build/update-site \
+     *   -categoryDefinition file:<project dir>/build/update-site/site.xml
+     * </code>
+     */
+    val updateSiteCategoryPublishing by registering {
+        dependsOn(updateSiteFeaturesAndBundlesPublishing)
+        doLast {
+            with(saros.gradle.eclipse.CategoryPublisher("4.8.0")) {
+                metadataRepository(project.file(updateSiteDirPath))
+                categoryDefinition(project.file("$updateSiteDirPath/site.xml"))
+                runUsingPdeInstallation()
+            }
+        }
+    }
+
+    /* Step 3 of update-site creation
+     *
+     * The creation-process is already completed after
+     * step 2, but this task removes the meta-data which
+     * were necessary for update-site creation but are
+     * not part of the update-site.
+     */
+    val updateSite by registering(Delete::class) {
+        dependsOn(updateSiteCategoryPublishing)
+
+        delete("$updateSiteDirPath/features/feature.xml")
+        delete(fileTree("$updateSiteDirPath/plugins") {
+            // Remove all bundles without version (which is appended during update-site build)
+            exclude("*_*.jar")
+        })
+        delete("$updateSiteDirPath/site.xml")
+    }
+
+    /* Creates a dropin at build/dropin
+     *
+     * Creates an update-site, removes the superfluous meta-data
+     * and creates a zip.
+     */
+    register("dropin", Zip::class) {
+        dependsOn(updateSite)
+
+        archiveFileName.set("saros-dropin.zip")
+        destinationDirectory.set(project.file("build/dropin"))
+
+        from(updateSiteDirPath) {
+            exclude("*.jar")
```

Comment:
Why does this exclude the jar files?

Reply:
I would have expected that the cleanup was already done in l.167 - l.176.

Reply:
The cleanup you mention removes all metadata that are not part of the update-site, but these metadata are part of the update-site, but not part of the dropin.